tools/utils.cc

```diff
@@ -69,6 +69,7 @@ void configure_tool_mode(app_template::seastar_options& opts, const sstring& log
     opts.reactor_opts.relaxed_dma.set_value();
     opts.reactor_opts.unsafe_bypass_fsync.set_value(true);
     opts.reactor_opts.kernel_page_cache.set_value(true);
+    opts.reactor_opts.reactor_backend.select_candidate("epoll");
```

Comment:
It's a little sad in that if we ever get io_uring working, we lose it here. Of course, reactor backend choice doesn't matter at all for tools, it's just cosmetic.

Reply:
(also seastar applications shouldn't know about reactor backends, they can disappear and reappear, but in practice there's no problem)

Reply:
Note that I already merged this PR, but we can always ask @denesb to send a followup patch.

Reply:
> Another option is to make Seastar smarter about choosing the backend without complaining too much when making the choice. I.e., if our default is aio but we can't manage to use aio because we're out of buffers or whatever, we could fall back to the second-best default (epoll), and do all of this silently.

By the way, what I wrote here would be a generally useful feature for **Seastar**, not just Scylla tools. In https://github.com/scylladb/seastar/issues/66 I even wanted to be able to fall back even further - to using threads for AIO. But I guess that falling back to something we already have - like epoll - should be easier.

Reply:
The patch is fine, we shouldn't waste time on a cosmetic issue.

io_uring doesn't need anything, but is currently disabled because it's slower than aio.

Reply:
Making seastar automatically drop aio is a good idea, but not easy, because one needs to pass around unrelated parameters.

Reply:
Maybe each reactor implementation could have an initialization function which tries to pre-allocate what it needs, and if this preallocation fails, we we don't choose this reactor and move to the next choice.
I agree that if we want to change reactor after we already started to use another one, everything will be harder and uglier.

Reply:
it does not use system wide aio slots. it uses per-process ring buffer.

Reply:
@nyh indeed that would be the best thing to do. But since performance is not that important here, the next-best thing (this patch) is also good enough, and is certainly much-much simpler. Hence I chose this.

Reply:
> @tchaikov I don't understand the details. In the commit message, Botond claims that the tool fails to allocate some "AIO blocks" and fails. What I suggested was that we try this allocation early, and if faili to allocate these block we simply don't use the aio backend and fall back to the next backend in the list.

sorry for the confusion. was trying to reply to Botond's question of 

> Does io_uring also require these AIO blocks? If not, we can use it for tools.

actually i proposed a similar fix at https://github.com/scylladb/scylladb/issues/17412.  but i think this fix is simpler. and it just works.